File: log_analysis/modules/logging/logger.py
from enum import Enum

class LogLevel(Enum):
    Error = 0,
    Warning = 1,
    Debug = 2

    def __int__(self):
        return self.value[0]

    def __str__(self):
        if self.value == LogLevel.Error:
            return "Error"
        elif self.value == LogLevel.Warning:
            return "Warning"
        elif self.value == LogLevel.Debug:
            return "Debug"
        else:
            return "Unknown log level"

class Logger:
    def __init__(self, max_log_level: LogLevel, callback):
        self.max_log_level = max_log_level
        self.callback = callback

    def log(self, level: LogLevel, message):
        # if self.callback is not None and int(level) <= int(self.max_log_level):
        if self.callback:
            self.callback(level, message)

File: log_analysis_web_backend/log_analysis/utils/src/logger.cpp
#include <utils/logger.h>

Logger& Logger::Get() {
    static Logger logger;
    return logger;
}

void Logger::SetSettings(LogLevel max_log_level, LogFunction log_function) {
    max_log_level_ = max_log_level;
    log_function_ = log_function;
}

void Logger::Write(LogLevel level, const char* message) {
    if (log_function_ && level <= max_log_level_) {
        log_function_(level, message);
    }
}

void Logger::Write(LogLevel level, const std::string& message) {
    Write(level, message.c_str());
}
